src/lib/forms-validation/common.ts

```diff
@@ -98,3 +98,88 @@ export function validateElementNamesAcrossNestedElements(
   }
   return elementNames
 }
+
+export function validateFormWorkflowMappingElements({
+  mappings,
+  validatedFormElements,
+  propertyName,
+}: {
+  mappings: SubmissionEventTypes.FormWorkflowEventElementMapping<
+    Record<string, unknown>
+  >[]
+  validatedFormElements: FormTypes.FormElement[]
+  propertyName: string
+}) {
+  for (let mappingIndex = 0; mappingIndex < mappings.length; mappingIndex++) {
+    const mapping = mappings[mappingIndex]
+    if (
+      mapping.type === 'FORM_ELEMENT' ||
+      mapping.type === 'FORM_FORM_ELEMENT'
+    ) {
+      const element = formElementsService.findFormElement(
+        validatedFormElements,
+        ({ id }) => id === mapping.formElementId,
+      )
+      if (!element) {
+        throw new Error(
+          `"${propertyName}[${mappingIndex}].formElementId" (${mapping.formElementId}) does not exist in "elements".`,
+        )
+      }
+      if (mapping.type === 'FORM_FORM_ELEMENT' && element.type !== 'form') {
+        throw new Error(
+          `"${propertyName}[${mappingIndex}].formElementId" (${mapping.formElementId}) must be the "id" for a "form" type element.`,
+        )
+      }
+    }
+  }
+}
+
+export function validatePDFConfiguration({
+  pdfConfiguration,
+  propertyName,
+  customPDFs,
+  validatedFormElements,
+}: {
+  pdfConfiguration: SubmissionEventTypes.PDFConfiguration | undefined
+  propertyName: string
+  customPDFs: FormTypes.Form['customPDFs']
+  validatedFormElements: FormTypes.FormElement[]
+}) {
+  if (!pdfConfiguration) {
+    return
+  }
+
+  const customPDF = pdfConfiguration.customPDF
+  if (customPDF) {
+    if (customPDF && !customPDFs?.some(({ id }) => id === customPDF.pdfId)) {
```

Comment:
```suggestion
    if (!customPDFs?.some(({ id }) => id === customPDF.pdfId)) {
```

Since we have already checked `customPDF` above